scripts/pin-external-packages.sh

```diff
@@ -10,3 +10,5 @@ for pkg in $PACKAGES; do
     echo "Pinning package" $pkg
     opam pin -y add src/external/$pkg
 done
+
+opam pin -y add src/app/check_opam_switch
```

Comment:
why pinning it btw? And I thought we didn't need this script anymore?

Reply:
The pin is there to install the package (since it's a local one) and have the binaries in the PATH.

About the `pin-external-packages.sh` script, PR #11388 will remove some of the sub-modules but it was decided to keep the ones which are often modified.

So except in the pure nix build, this script still seems needed. And I did not managed to specify path-pinned packages in the `opam.export` file without using an absolute path (so it cannot be shared).
This could be an argument for generating the `opam.export` file.